extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -1299,6 +1299,9 @@ export class AmpStoryPage extends AMP.BaseElement {
     if (this.isAd()) {
       distance = Math.min(distance, 2);
     }
+    if (distance == this.getDistance()) {
```

Comment:
Please ensure that `preloadAllMedia_` is not called twice, a simple boolean check is fine.

Reply:
Without getting into too much context, we need to call `preloadAllMedia_` multiple times. There's an early return two methods further in the execution stack, but it's used to re-allocate video elements if they've been evicted